doc/user-guide.md

```diff
@@ -423,6 +423,87 @@ To implement complex deletion logic, you can add a finalizer to your Custom Reso
 [official Kubernetes documentation on finalizers](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers).
 
+**Example:**
+
+Following an example of your Custom Resource with. 
+
+```go
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  name: example-memcached
+  finalizers:
+  - finalizer.cache.example.com
+spec:
+  # Add fields here
+  size: 3
+```
+
+Following and example of the controller implemented to work with. 
+
+```go
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+    reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+    reqLogger.Info("Reconciling Memcached")
+        
+    // Fetch the Memcached instance
+    memcached := &cachev1alpha1.Memcached{}
+    err := r.client.Get(context.TODO(), request.NamespacedName, memcached)
+
+    //{...} reconcile implementations ...
+    
+    //Perform the actions required to allow the CR deletion
+    if err := r.doFinalizerActions(memcached); err != nil {
+        reqLogger.Error(err, "Unable to perform actions to finalize the Mencached", "Memcached.Namespace", memcached.Namespace, "Memcached.Name", memcached.Name)
+        return reconcile.Result{}, err
+    }
+    
+    //Check Finalizer
+    if err := r.checkFinalizer(memcached); err != nil {
+        reqLogger.Error(err, "Unable to satisfy finalizer conditionals for Mencached", "Memcached.Namespace", memcached.Namespace, "Memcached.Name", memcached.Name)
+        return reconcile.Result{}, err
+    }
+    
+    return reconcile.Result{}, nil
+}
+
+//checkFinalizer returns error when was not possible check that the defined actions were performed with success
+func (r *ReconcileMemcached) checkFinalizer(m *cachev1alpha1.Memcached) error {
+	if len(m.GetFinalizers()) > 0 && m.GetDeletionTimestamp() != nil {
+		// Check if finalizer conditional was performed with sucess
+		if  isFinalizerActionMet(m) {
+			return fmt.Errorf("Unable to met criteria for Mancached deletion.")
+		}
+		
+		// Remove finalizer and deletion timestamp from the CR to allow cluster delete it
+		m.SetFinalizers(nil)
+		m.SetDeletionTimestamp(nil)
+		
+		// Update CR
+		err := r.client.Update(context.TODO(), m)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+//doFinalizerActions perform the actions required before the CR be deleted
+func (r *ReconcileMemcached) doFinalizerActions(m *cachev1alpha1.Memcached) error {
+	// Check if the Memcached CR has a finalizer defined
+	if len(m.GetFinalizers()) > 0 && m.GetDeletionTimestamp() != nil {
+		//TODO: Add the impl that is required to allow the CR be deleted
+	}
+	return nil
+}
+
+//isFinalizerActionMet returns true when the criteria to allow the the CR be deleted was attended
+func isFinalizerActionMet(m *cachev1alpha1.Memcached) bool {
+	//TODO: Add the impl which will check if the actions to finalize this CR was performed with success
```

Comment:
```suggestion
	// TODO: Add the logic which will check if the actions to finalize this CR was performed with success
```

Again is this the TODO left for the user, then we need to add the user after TODO